src/main/java/reposense/model/RepoConfiguration.java

```diff
@@ -105,14 +106,22 @@ public static void merge(List<RepoConfiguration> repoConfigs, List<AuthorConfigu
                 continue;
             }
 
-            matchingRepoConfig.setAuthorConfiguration(authorConfig);
+            if (!isDefaultBranch) {
```

Comment:
This check should be combined into `getMatchingRepoConfig` itself.
Reason: the entire purpose of this foreach block is to validate and perform the following line.
The logger warning message above can then be updated to match the new logic

Reply:
Sorry I don't really get this comment. So you want something like, if the `AuthorConfig` use default branch, `getMatchingRepoConfig` should not return any matching repo? The check here is actually used to prevent overwriting of `AuthorConfig` if one of them uses default brance.

Reply:
>So you want something like, if the AuthorConfig use default branch, getMatchingRepoConfig should not return any matching repo?

If this is intended to be part of the check, then I would say yes.
The thing is, now you have 2 separate checks that leads to the final conclusion which is 
```
                matchingRepoConfig.setAuthorConfiguration(authorConfig);
```
They should be ideally together, otherwise the `matchingRepoConfig` doesn't really serve much purpose if you're not going to use it in the end `if (!isDefaultBranch)`.


Coming back to the main topic (which I realized that I didn't asked earlier), given that you have only have an author config that is using default branch value, does that means that it won't be set/bind to any `RepoConfiguration` in this case?

Reply:
No, it will bind to all `RepoConfig` that has the same location

Reply:
It doesn't seems that you're performing `setAuthorConfiguration` if the author config is using default branch. In that case the ignore glob list is not being propagate to the authors.

Can you revise your PR? Also put up the test case for the scenario regarding the issue with ignore glob list.

Reply:
When using default branch, the `ignoreGlobList` is propagated via the method `matchingRepoConfig.addAuthors(authorConfig.getAuthorList())` in the code below. The `addAuthors` calls `addAuthor` in `AuthorConfig` which calls `propagateIgnoreGlobList`. The test case now already check for this issue.

Reply:
>When using default branch, the ignoreGlobList is propagated via the method matchingRepoConfig.addAuthors(authorConfig.getAuthorList()) in the code block below.

I see. With this new implementation, it doesn't quite make sense to split the process into 2 groups anymore; it become quite confusing. 

Can you combine and restructure the logic such that,

if location empty -> add to all repos, continue
find all matched location repos
if matched location repos is empty -> print repo not found, continue
if branch is default -> add to all matched location repos, continue
find all matched branch repos
if branch matches -> add to the repo, continue
print repo branch not found

>The test case now already check for this issue.

The test case did not include any ignore globs, can you add some to ensure that the functionality is as of expected?

Reply:
The reason for separating the loops is because now for matched repo we use `setAuthorConfiguration` which saves the time adding the authors into authorList in `AuthorConfig` one by one. If we don't complete all the matched repo first, the `setAuthorConfiguration` will overwrite the authorlist data of empty location/default branch ones.

Reply:
> The test case did not include any ignore globs, can you add some to ensure that the functionality is as of expected?

The `merge_twoRepoConfigs_success` in `CSVParserTest` uses a authorconfig.csv in which the second author using default branch without any ignoreGlobList, while the repoconfig.csv contains 2 repo that will be mapped to the default branch author, and the repo all have ignoreGlobList `Collated**`. The expected repo contains expected second author having ignoreGlobList Collated**. And the expected repo is compared to actual repo at the end using `TestUtil.compareRepoConfig`, which checks if the author having same ignoreGlobList (if the colated** is propagated to authorlist)

Reply:
>The reason for separating the loops is because now for matched repo we use setAuthorConfiguration which saves the time adding the authors into authorList in AuthorConfig one by one. If we don't complete all the matched repo first, the setAuthorConfiguration will overwrite the authorlist data of empty location/default branch ones.

Yes, but with the additional logics, it is no longer ideal to separate the logic anymore; it hinders the readability.

 >The merge_twoRepoConfigs_success in CSVParserTest uses a authorconfig.csv in which the second author using default branch without any ignoreGlobList, while the repoconfig.csv contains 2 repo that will be mapped to the default branch author, and the repo all have ignoreGlobList Collated**. The expected repo contains expected second author having ignoreGlobList Collated**. And the expected repo is compared to actual repo at the end using TestUtil.compareRepoConfig, which checks if the author having same ignoreGlobList (if the colated** is propagated to authorlist)

Yes, but without any ignore globs in the author configuration, can you really be sure that it is propagating and not overriding?